custom_components/opensprinkler/__init__.py

```diff
@@ -187,6 +187,7 @@ def device_info(self):
             "identifiers": {(DOMAIN, slugify(self._entry.unique_id))},
             "name": self._name,
             "manufacturer": "OpenSprinkler",
+            "configuration_url":  f"http://{controller.ip_address}",            
```

Comment:
Instead of using the controller details here from the API, I think we should be using the URL from the HA configuration, `self._entry.data.get(CONF_URL)` works here.

This means it will have the protocol included and it will work for those that use a proxy or similar